src/Data/Codec.purs

```diff
@@ -88,57 +88,73 @@ bihoistGCodec f g (GCodec dec (Star h)) = GCodec (f dec) (Star (g <<< h))
 -- | ```
 infixl 5 lcmap as ~
 
-type Codec m a b c d = GCodec (ReaderT a m) (Writer b) c d
+type Codec decoderF decoderConfig encoderAccum encoderInputType decoderTargetType = GCodec (ReaderT decoderConfig decoderF) (Writer encoderAccum) encoderInputType decoderTargetType
```

Comment:
I actually didn't understand why monad for the encoder is `Writer`

(NOTE: in `Writer w a` the `w` is what is actually returned by encoder, not `a`) 

I first thought that `a` should be returned, but it's not, the `w` is returned

Then I thought that monoidal property of `encoderAccum` should be used, but it's not

Q1: What is the meaning `a` in `Writer` for encoder? Does it play role of storage for temprorary results?

Q2: Maybe there exists some type like `Writer` where `w` is not a monoid to make clear that `w` monoidal property is not used?

Reply:
@garyb